pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -189,3 +178,143 @@ func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	}
+
+	var secret *corev1.Secret
+	create := false
+
+	if userSecret == nil {
+		action = CreatePullSecret
+		create = true
+	}
+
+	// userSecret can happen to have broken type, which means it have to be recreated
+	// with proper type
+	if userSecret != nil && (userSecret.Type != corev1.SecretTypeDockerConfigJson || userSecret.Data == nil) {
+		action = RecreatePullSecret
+		create = true
+	}
+
+	if create {
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pullSecretName.Name,
+				Namespace: pullSecretName.Namespace,
+			},
+			Type: corev1.SecretTypeDockerConfigJson,
+		}
+		secret.Data = make(map[string][]byte)
+	} else {
+		secret = userSecret.DeepCopy()
+		if !json.Valid(secret.Data[corev1.DockerConfigJsonKey]) {
+			delete(secret.Data, corev1.DockerConfigJsonKey)
+		}
+	}
+
+	fixedData, update, err := pullsecret.Merge(string(secret.Data[corev1.DockerConfigJsonKey]), string(operatorSecret.Data[corev1.DockerConfigJsonKey]))
+	if err != nil {
+		return nil, NoAction, err
+	}
+
+	if !create && !update {
+		return secret, NoAction, nil
+	}
+
+	if !create && update {
+		action = UpdatePullSecret
+	}
+
+	secret.Data[corev1.DockerConfigJsonKey] = []byte(fixedData)
+	return secret, action, nil
+}
+
+// emitGlobalPullSecretChange performs update of the secret when the user pull secret is broken or require recreation
+func (r *PullSecretReconciler) emitGlobalPullSecretChange(ctx context.Context, secret *corev1.Secret, action PullSecretAction) error {
+	if action == NoAction {
+		// no action required
+		return nil
+	}
+
+	if action == RecreatePullSecret {
+		// unfortunately the type field is immutable.
+		err := r.kubernetescli.CoreV1().Secrets(secret.Namespace).Delete(ctx, secret.Name, metav1.DeleteOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	if action == CreatePullSecret || action == RecreatePullSecret {
+		r.log.Info("re-creating pull secret")
+		_, err := r.kubernetescli.CoreV1().Secrets(secret.Namespace).Create(ctx, secret, metav1.CreateOptions{})
+
+		return err
+	}
+
+	r.log.Info("updating pull secret")
+	_, err := r.kubernetescli.CoreV1().Secrets(secret.Namespace).Update(ctx, secret, metav1.UpdateOptions{})
+	return err
+}
+
+func (r *PullSecretReconciler) updateRedHatKeyCondition(secret *corev1.Secret) (*status.Condition, error) {
+	// parse keys and validate JSON
+	parsedKeys, err := pullsecret.UnmarshalSecretData(secret)
+	foundKey := false
+	failed := false
+	if err != nil {
+		r.log.Info("pull secret is not valid json - recreating")
+		failed = true
+	} else {
+		foundKey = r.checkRHRegistryKey(parsedKeys)
+	}
+
+	keyCondition := r.keyCondition(failed, foundKey)
+	return keyCondition, nil
+}
+
+// checkRHRegistryKey checks whether the rhRegistry key:
+//   - redhat.registry.io
+// is present in the pullSecret
+func (r *PullSecretReconciler) checkRHRegistryKey(psData *pullsecret.SerializedAuthMap) (foundKey bool) {
+	rhKey := "registry.redhat.io"
```

Comment:
You had code before where it was detecting ALL keys and putting them into Condition. Please do that. And make these vars configurable for controller level.

Reply:
Other keys gives us information too like if telemeter is enabled, samples enabled, cloud.redhat.com enabled.